docarray/typing/tensor/abstract_tensor.py

```diff
@@ -91,3 +91,9 @@ def validate(
     def __class_getitem__(cls, item: Any):
         target_shape = cls.__validate_getitem__(item)
         return cls._create_parametrized_type(target_shape)
+
+    @classmethod
+    @abc.abstractmethod
+    def __docarray_stack__(cls: Type[T], seq: Union[List[T], Tuple[T]]) -> T:
+        """Stack a sequence of tensors into a single tensor."""
```

Comment:
Is the point of this that it ensures that the return is an instance of our custom class and not the base class?